packages/code-utils/test/index.test.ts

```diff
@@ -17,21 +19,43 @@ describe("CodeUtils.parseCode", function () {
     const parsedCode = CodeUtils.parseCode("[]");
     assert(parsedCode);
     assert(Array.isArray(parsedCode));
-    assert(parsedCode.length === 0);
+    assert.equal(parsedCode.length, 0);
+  });
+
+  it("Strips metadata", function () {
+    const bytecode =
+      "0x608060405200fea165627a7a7230582026bee5eca148486641ec418964c5d4d663105706cf51bf8b7130a50412a5660d0029";
+    const parsedCode = CodeUtils.parseCode(bytecode, {
+      attemptStripMetadata: true
+    });
+    assert(parsedCode);
+    assert.equal(parsedCode.length, 5);
   });
 
   it("doesn't strip invalid metadata", function () {
-    const parsedCode = CodeUtils.parseCode(
-      "0x600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600060000100"
-    );
+    const bytecode =
+      "0x600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600050600060000100";
```

Comment:
I don't think that you necessarily need to fix this as part of this PR because this problem is quite pervasive in our tests, but I think tests like this would be a lot easier for people such as myself to understand these tests if the code that generated these hex strings was checked in somewhere and referenced by this test suite.

To be clear I think this bytecode is generated code that _should_ be checked in, and I don't think that we should take the extra performance or reliability hit to regenerating this bytecode on every run of the test suite. So take the word "referenced" above to have a very loose definition - even a comment pointing at a script that generates the bytecode from a source file would be fine.

Reply:
Though now that I wrote that, I think I might've commented on a hex string that would've needed to be hand-edited, as it's the invalid case, but the point still remains. :-)

Although even in this case it'd be good to have it be mutating a different correct bytestring so that it's clear _how_ the metadata here is invalid.

Reply:
Oh, that's an interesting idea!

Some of these tests predate me and I simply don't know what the source code is that generated them.  And as for the ones I generated... I'm not sure I know just which source code of mine I generated all those of those from, either!

(The one you commented on I actually hand-wrote... not just hand-edited, hand-wrote. :)  It just repeatedly pushes a zero onto the stack and then pops it off; finally it pushes two zeroes onto the stack, adds them, and then terminates!  I don't think any of the ones made by compiling actual source code have any edits in them.)

I agree actually having the source code on hand would be nice.  That said, compiling Vyper in Truffle does have some complications, especially if you need to do so on a specific version of Vyper such as 0.3.4, especially especially if you maybe want multiple specific versions, as we do here.  (The "long input" test was compiled on a pre-0.3.4 version of Vyper, and that's essential to the test.)  So unfortunately I don't know if that's practical.

But yeah I could add some sort of comment at least... where would the actual source file go, though?  (For the ones whose source files we can determine, anyway.)

Reply:
Again, I think this is fine to merge as-is, however I think we need to raise an issue about this with some priority, and at some stage (pending team consensus) we should probably draw a line in the sand and prohibit new tests that contain opaque/unreadable binaries.

I'll go ahead and create that issue (assuming one hasn't been created already).

Reply:
Issue created as #5419